src/main/org/codehaus/groovy/reflection/CachedMethod.java

```diff
@@ -90,6 +92,12 @@ public CachedClass getDeclaringClass() {
 
     public final Object invoke(Object object, Object[] arguments) {
         try {
+            AccessPermissionChecker.checkAccessPermission(cachedMethod);
+        }
+        catch (AccessControlException ex) {
+            throw new InvokerInvocationException(new GroovyRuntimeException("Illegal access to method" + cachedMethod.getName()));
+        }
+        try {
             return cachedMethod.invoke(object, arguments);

```

Comment:
I think that InvokerInvocationException is specific to call of invoke() and it should be not thrown  by AccessPermissionChecker directly.